tests/test_scripts.py

```diff
@@ -0,0 +1,588 @@
+    args = create_dummy_args(
+        component_repository="https://github.com/fepitre/qubes-core-qrexec",
+        component_directory=temp_directory,
+        git_branch="builderv2-tests-1",
+        fetch_versions_only=True,
+    )
+    with pytest.raises(ValueError) as e:
+        get_and_verify_source(args)
+    (msg,) = e.value.args
+    assert "No version tag." == msg
+
+
+def test_existing_repository_no_version_tags(capsys, temp_directory):
+    component_repository = "https://github.com/fepitre/qubes-core-qrexec"
+    git_branch = "builderv2-tests-1"
+    subprocess.run(
+        ["git", "clone", "-b", git_branch, component_repository, temp_directory],
+        check=True,
+        capture_output=True,
+    )
+    args = create_dummy_args(
+        component_repository=component_repository,
+        component_directory=temp_directory,
+        git_branch=git_branch,
+        fetch_versions_only=True,
+    )
+    get_and_verify_source(args)
+    assert capsys.readouterr().out == "No version tag.\n"
+    assert not (temp_directory / ".git/FETCH_HEAD").exists()
+
+
+def test_existing_repository_with_version_tags(capsys, temp_directory):
+    component_repository = "https://github.com/fepitre/qubes-core-qrexec"
+    git_branch = "builderv2-tests-2"
+    subprocess.run(
+        ["git", "clone", "-b", git_branch, component_repository, temp_directory],
+        check=True,
+        capture_output=True,
+    )
+    args = create_dummy_args(
+        component_repository=component_repository,
+        component_directory=temp_directory,
+        git_branch=git_branch,
+        fetch_versions_only=True,
+    )
+    get_and_verify_source(args)
+    assert (
+        capsys.readouterr().out
+        == "--> Verifying tags...\n---> Good tag 83fc687c10da7ca5625b13d4d1fadd400748c427.\n---> Good tag 76d34269f7910a17f186c4f819583faae404bb83.\n---> Good tag 644dd42e21b788abb7483c1ec0d4ab3a7d9d20aa.\nEnough distinct tag signatures. Found 1, mandatory minimum is 1.\n--> Merging...\n"
+    )
+    assert (temp_directory / ".git/FETCH_HEAD").exists()
+
+
+def test_existing_repository_with_switching_branch(capsys, temp_directory):
+    component_repository = "https://github.com/qubesos/qubes-core-qrexec"
+
+    # Manually clone the repo
+    subprocess.run(
+        ["git", "clone", component_repository, temp_directory],
+        check=True,
+        capture_output=True,
+    )
+
+    # Ensure we are not on a version tag
+    subprocess.run(
+        ["git", "checkout", "53351ebf68d88e2fee06c761a9d5b1669b5713fc"],
+        check=True,
+        capture_output=True,
+        cwd=temp_directory,
+    )
+
+    # Fetch latest version
+    args = create_dummy_args(
+        component_repository=component_repository,
+        component_directory=temp_directory,
+        fetch_versions_only=True,
+    )
+    get_and_verify_source(args)
+
+    # Check if we have a version tag on HEAD
+    vtag = subprocess.run(
+        [
+            "git",
+            "describe",
```

Comment:
Describe (especially with `--abrev=0`) will find a tag, even if not on top. Here you want `git tag --points-at=HEAD v*` instead.

Reply:
Done.